pkg/apis/cluster/v1alpha1/common_types.go

```diff
@@ -41,3 +41,27 @@ type ProviderConfig struct {
 type ProviderConfigSource struct {
 	// TODO(roberthbailey): Fill these in later
 }
+
+// ProviderStatus defines the provider-specific observed state.
+type ProviderStatus struct {
+
+	// No more than one of the following may be specified.
+
+	// Value is an inlined, serialized representation of the resource
+	// configuration. It is recommended that providers maintain their own
+	// versioned API types that should be serialized/deserialized from this
+	// field, akin to component config.
+	// +optional
+	Value *runtime.RawExtension `json:"value,omitempty"`
+
+	// Source for the provider status. Cannot be used if value is
+	// not empty.
+	// +optional
+	ValueFrom *ProviderStatusSource `json:valueFrom,omitempty`
+}
+
+// ProviderStatusSource represents a source for the provider-specific
+// resource configuration.
+type ProviderStatusSource struct {
+	// TODO(roberthbailey): Fill these in later
```

Comment:
please use your username for the TODO unless robertbailey is aware that you created another TODO for him

Reply:
Good point. That had crossed my mind. I left it there because I thought whatever he wanted to do for ProviderConfig would also apply to ProviderSource. I'll reach out to him right now :)

Reply:
@roberthbailey Should I put my name on this TODO? I left your name on the TODO because I thought whatever work you planned to do for `ProviderConfigSource` would also apply to `ProviderStatusSource`.

Reply:
@roberthbailey You're absolutely right; there's no use case for a shared ProviderStatus. I should have ProviderStatusSource to begin with. Fixing.